nix/pkgs/haskell/haskell.nix

```diff
@@ -59,9 +58,138 @@ let
+      ({ pkgs, ... }: lib.mkIf pkgs.stdenv.hostPlatform.isWindows {
+        nonReinstallablePkgs =
+          [
+            "rts"
+            "ghc-heap"
+            "ghc-prim"
+            "integer-gmp"
+            "integer-simple"
+            "base"
+            "deepseq"
+            "array"
+            "ghc-boot-th"
+            "pretty"
+            "template-haskell"
+            # ghcjs custom packages
+            "ghcjs-prim"
+            "ghcjs-th"
+            "ghc"
+            "array"
+            "binary"
+            "bytestring"
+            "containers"
+            "filepath"
+            "ghc-compact"
+            "ghc-prim"
+            # "ghci" "haskeline"
+            "hpc"
+            "mtl"
+            "parsec"
+            "text"
+            "transformers"
+            "xhtml"
+            # "stm" "terminfo"
+          ];
+        packages.Win32.components.library.build-tools = lib.mkForce [ ];
+      })
+      ({ pkgs, ... }: lib.mkIf (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) {
+        packages = {
+          # Things that need plutus-tx-plugins
+          marlowe.package.buildable = false; # Would also require libpq
+          marlowe-actus.package.buildable = false;
+          marlowe-dashboard-server.package.buildable = false;
+          marlowe-playground-server.package.buildable = false; # Would also require libpq
+          marlowe-symbolic.package.buildable = false;
+          playground-common.package.buildable = false;
+          plutus-benchmark.package.buildable = false;
+          plutus-chain-index.package.buildable = false;
+          plutus-contract.package.buildable = false;
+          plutus-errors.package.buildable = false;
+          plutus-ledger.package.buildable = false;
+          plutus-pab.package.buildable = false;
+          plutus-playground-server.package.buildable = false; # Would also require libpq
+          plutus-use-cases.package.buildable = false;
+          web-ghc.package.buildable = false;
+          # Needs agda
+          plutus-metatheory.package.buildable = false;
+          # These need R
+          plutus-core.components.benchmarks.cost-model-test.buildable = lib.mkForce false;
+          plutus-core.components.benchmarks.update-cost-model.buildable = lib.mkForce false;
+          # Windows build of libpq is marked as broken
+          fake-pab.package.buildable = false;
+        };
+      })
+      ({ pkgs, ... }:
+        let
+          symlinkDlls = ''
+            ln -s ${libsodium-vrf}/bin/libsodium-23.dll $out/bin/libsodium-23.dll
+            ln -s ${pkgs.buildPackages.gcc.cc}/x86_64-w64-mingw32/lib/libgcc_s_seh-1.dll $out/bin/libgcc_s_seh-1.dll
+            ln -s ${pkgs.buildPackages.gcc.cc}/x86_64-w64-mingw32/lib/libstdc++-6.dll $out/bin/libstdc++-6.dll
+            ln -s ${pkgs.windows.mcfgthreads}/bin/mcfgthread-12.dll $out/bin/mcfgthread-12.dll
+          '';
+        in
+        lib.mkIf (pkgs.stdenv.hostPlatform.isWindows) {
```

Comment:
can we merge this with the other `hostPlatform.isWindows` module? So we have one windows module, one cross module, and one unconditional module

Reply:
Turns out we don't need the module to make Win32 reinstallable any more.